### PR TITLE
Chore/drop cheerio

### PR DESCRIPTION
**Pull request by MrAdex77**
## Summary by CodeRabbit

* **New Features**
  * Added HTML-to-plain-text conversion for app descriptions, including decoding supported entities, stripping HTML tags, and normalizing line breaks.
* **Bug Fixes**
  * Improved robustness for malformed HTML and invalid/out-of-range character references.
  * Ensured generated description text matches legacy output, including consistent newline behavior.
* **Tests**
  * Added unit tests for conversion behavior and entity/markup edge cases.
  * Added end-to-end “live parity” checks against a legacy text pipeline.
* **Documentation**
  * Refined README comparison table spacing/formatting.

### CODE REVIEW

File: src/core/htmlText.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    return decodeNumericReference(Number.parseInt(body.slice(1), 10));
+  }
+  return NAMED_ENTITIES[body] ?? match;
+}
+
+export function htmlToPlainText(html: string): string {
+  return html
+    .replace(BR_TAGS, '\r\n')
+    .replace(CARRIAGE_RETURNS, '\n')
+    .replace(TAGS, '')
```
[the remark this answers is not in the document]
Comment (MrAdex77) [reply src/core/htmlText.ts]: Reworked in 78c6cee: tag stripping now iterates the replace to a fixpoint, so no complete tag (including `<script`) can survive stripping; a unit test pins the nested-angle-bracket scenario. Output is behavior-identical, guarded by the 49-case cheerio parity suite.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/core/htmlText.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+  if (body.startsWith('#')) {
+    return decodeNumericReference(Number.parseInt(body.slice(1), 10));
+  }
+  return NAMED_ENTITIES[body] ?? match;
+}
+
+function stripTags(html: string): string {
+  let previous = html;
+  let stripped = html.replace(TAGS, '');
```
[the remark this answers is not in the document]
Comment (MrAdex77) [reply src/core/htmlText.ts]: Fixed in 54ab6ba: the loop now feeds each replace result back through the same call until a fixpoint, the dataflow shape CodeQL recognizes as complete sanitization; the CodeQL check passes on this head. Output is unchanged, guarded by the cheerio parity suite and a nested-angle-bracket unit test.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
